openedx/core/djangoapps/courseware_api/serializers.py

```diff
@@ -56,6 +56,30 @@ class Meta:
         ref_name = 'courseware_api'
 
 
+class CourseProgramSerializer(serializers.Serializer):
+    progress = serializers.SerializerMethodField()
+    title = serializers.CharField()
+    type = serializers.CharField()
+    url = serializers.SerializerMethodField()
```

Comment:
Look up AbsoluteURLField. It's a hack we wrote, but IIRC, it does what you want in your serializer method.